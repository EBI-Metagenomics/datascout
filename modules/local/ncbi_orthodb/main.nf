process NCBI_ORTHODB {
    maxForks 3

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/d2/d2cc550ff67f8541d44dc2db1b5d2d2e1cfccfe8536222b49788deefde7460f0/data' :
        'community.wave.seqera.io/library/python_pip_biopython_requests:725bda83fb97ec48' }"

    label "process_medium"

    tag "${meta}"

    input:
      tuple val(meta), path(tax_ranks), val(max_rank)
      val(max_clusters)

    output:
      tuple val(meta), path("${meta.id}_orthodb_dir"), emit: orthodb_results
      path("versions.yml"), emit: versions

    script:
    def max_clusters_arg = max_clusters ? "--max_clusters ${max_clusters}" : ""
    """
    mkdir -p ${meta.id}_orthodb_dir
    ncbi_orthodb_data.py --tax_file ${tax_ranks} --lineage_max ${max_rank} --output "${meta.id}_orthodb_dir" ${max_clusters_arg}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
    \$( ncbi_orthodb_data.py --version 2>&1 )
      Python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """
}
