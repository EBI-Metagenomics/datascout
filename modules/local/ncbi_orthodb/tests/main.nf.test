#https://github.com/nf-core/modules/blob/master/modules/nf-core/goat/taxonsearch/tests/main.nf.test
