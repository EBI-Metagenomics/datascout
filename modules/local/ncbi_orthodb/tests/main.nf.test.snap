{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "test_sample1",
                        "genome_id": "GCA_003719455.1",
                        "ena_tax": "default"
                    },
                    "/Users/vkale/Downloads/pipelines/datascout/.nf-test/tests/90f092ce12859450d7a2c9a5116c66b0/work/ae/0cb46e05b3bcd550639bf8c991eac1/test_sample1_orthodb_dir"
                ]
            ],
            [
                "/Users/vkale/Downloads/pipelines/datascout/.nf-test/tests/90f092ce12859450d7a2c9a5116c66b0/work/ae/0cb46e05b3bcd550639bf8c991eac1/versions.yml"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-09-09T12:39:58.308941"
    }
}